### ISSUE TITLE
Mimetype .hamlc is not getting registered

### ISSUE THREAD

**Issue by vs37559** [emilioforrer/haml_coffee_assets#182]
I have created a new Rails application with `Ruby v3.0.2`, `Rails 6.1.4.1` and `Sprockets v4.0.2`
When I try to require my `.jst.hamlc` template in `application.js`

Following error shows up during precompile:
`Sprockets::FileNotFound: couldn't find file 'settings/components/integrations/integration_layout_template' with type 'application/javascript'
`
I checked for Registered Mime but there is no entry for hamlc in the list.

**Comment by mathieujobin** [comment]
Did you find out if this was indeed a problem with the gem ? Are you able to provide a test to reproduce it ? 
I'm afraid I won't be able to look at your problems, but if you open a PR with a fix. I can review merge and release.

**Comment by vs37559** [comment]
@mathieujobin Yes, It is a problem with the gem. With latest sprockets extension chaining is no longer supported and extensions like `.jst.hamlc` are needed to be explicitly registered. I don't have a PR for this since Sprockets has changed its way of compiling and are not relying on Tilt anymore( is what I believe). Will you be able to check this as I see some activity regarding the support for Rails 7? Maybe this also could be included in the same package?

**Comment by mathieujobin** [comment]
We are using this gem with rails 7, and we did need to rename some view files. I'm not 100% how related it is to your problems.
but we had view files like `_name.mustache.slim` that could not be loaded.
we just renamed them `_name_mustache.slim` effectively replacing one dot for a underscore and it solved our problem.

**Comment by vs37559** [comment]
I have found a workaround as mentioned below i.e. by registering `text/hamlc` to `application/javascript+function` as Sprocket provides a JSTProcessor for `application/javascript+function => application/javascript`
``` ruby
# lib/haml_coffee_assets/rails/engine.rb:38
env.register_mime_type 'text/hamlc', extensions: ['.hamlc', '.jst.hamlc']
env.register_transformer 'text/hamlc', 'application/javascript+function', ::HamlCoffeeAssets::Transformer
```
With this I was able to counter the initially mentioned error in my Rails 7 (earlier mentioned as Rails 6) & Sprockets 4 project.
Is this something we can do moving forward?

**Comment by mathieujobin** [comment]
this should be fixed in 1.25.1 

please try it
